src/main/java/com/epam/esm/dao/GiftCertificateDao.java

```diff
@@ -2,67 +2,63 @@
 
 import com.epam.esm.entity.GiftCertificate;
 import com.epam.esm.sql.SqlQueries;
-import com.epam.esm.mapper.GiftCertificateMapper;
 import com.epam.esm.utils.QueryConstructor;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.jdbc.core.JdbcTemplate;
-import org.springframework.jdbc.support.GeneratedKeyHolder;
-import org.springframework.jdbc.support.KeyHolder;
-import org.springframework.stereotype.Component;
-import java.sql.PreparedStatement;
-import java.sql.Statement;
+import org.springframework.stereotype.Repository;
+import javax.persistence.EntityManager;
+import java.math.BigInteger;
 import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
-@Component
+@Repository
 public class GiftCertificateDao implements EntityDao<GiftCertificate> {
 
-    private final JdbcTemplate template;
-    private final GiftCertificateMapper mapper;
+    private final EntityManager manager;
     private final QueryConstructor constructor;
 
     @Autowired
-    public GiftCertificateDao(JdbcTemplate template, GiftCertificateMapper mapper, QueryConstructor constructor) {
-        this.template = template;
-        this.mapper = mapper;
+    public GiftCertificateDao(EntityManager manager, QueryConstructor constructor) {
+        this.manager = manager;
         this.constructor = constructor;
     }
 
     @Override
     public void deleteById(long id) {
-        template.update(SqlQueries.DELETE_CERTIFICATE, id);
+        manager.getTransaction().begin();
```

Comment:
Why do you need to explicitly begin & end a transaction?

Reply:
Not nedded